### PR TITLE
Merged IstioConfigList and ExportedResources into one.

### PR DESCRIPTION
**Pull request by hhovsepy**
RFE https://github.com/kiali/kiali/issues/4528

Now there is single IstioConfigList which retrieves data from Registry and contains all exported Istio Configs.

For QE:
Regression testing all validation scenarios.
UI/E2E regression tests.

Verify DestinationRule's subset label's cross-namespace functionality.
1. Install bookinfo, bookinfo2, bookinfo3.
2. in bookinfo, delete the deployment reviews-v1
3. in bookinfo2, delete the deployment reviews-v2
4.  in bookinfo3, delete the deployment reviews-v3
5. in bookinfo, in DR "reviews", leave the host default "reviews"
6. verify info message on subset v1 not exist.
7. in bookinfo, in DR "reviews", change the host to "reviews.bookinfo.svc.cluster.local"
8. verify info message on subset v1 not exist.
9. in bookinfo, in DR "reviews", change the host to "reviews.bookinfo2.svc.cluster.local"
10. verify info message on only subset v2 not exist.
11. in bookinfo, in DR "reviews", change the host to "reviews.bookinfo3.svc.cluster.local"
12. verify info message on only subset v3 not exist.

### CODE REVIEW

**Comment by lucasponce** [comment]
@hhovsepy I wonder if it's worth addressing the pending Istio config fetched in the validations and trying to consolidate it.

For example, I'm still seeing:

image

The "fetchNonLocalmTLSConfigs" and "fetchNonLocalmTLSConfigs" are just other calls to the business.IstioConfigList" that I think it can be fetched as part of the main "fetchIstioConfigList".

I ask because we are splitting in too many minor PRs this effort, and I wonder if that could also be a source of regressions.

If just a thought, we can keep working in small PRs but with the goal to consolidate all of these calls I think.

File: business/istio_validations.go [hunk 1]
```diff
@@ -63,23 +63,20 @@ func (in *IstioValidationsService) GetValidations(ctx context.Context, namespace
...
 
-	wg.Add(8) // We need to add these here to make sure we don't execute wg.Wait() before scheduler has started goroutines
+	wg.Add(6) // We need to add these here to make sure we don't execute wg.Wait() before scheduler has started goroutines
 
 	// We fetch without target service as some validations will require full-namespace details
-	go in.fetchIstioConfigList(ctx, &istioConfigList, namespace, errChan, &wg)
-	go in.fetchExportedResources(ctx, &exportedResources, namespace, errChan, &wg)
+	go in.fetchIstioConfigList(ctx, &istioConfigList, &mtlsDetails, &rbacDetails, namespace, errChan, &wg)
 	go in.fetchNamespaces(ctx, &namespaces, errChan, &wg)
 	go in.fetchAllWorkloads(ctx, &workloadsPerNamespace, errChan, &wg)
```
Comment (lucasponce) [inline business/istio_validations.go:78]: Another comment that I think it's worth reviewing in this work.
At this level with have a "fetchNamespaces" and a "fetchAllWorkloads", but inside the "fetchAllWorkloads" we are repeating the "GetNamespaces".

I think that's something we can review if we are not duplicating calls.

File: business/istio_validations.go [hunk 2]
```diff
@@ -157,42 +153,40 @@ func (in *IstioValidationsService) GetIstioObjectValidations(ctx context.Context
...
 
 	// Get all the Istio objects from a Namespace and all gateways from every namespace
-	wg.Add(9)
+	wg.Add(7)
 	go in.fetchNamespaces(ctx, &namespaces, errChan, &wg)
-	go in.fetchIstioConfigList(ctx, &istioConfigList, namespace, errChan, &wg)
-	go in.fetchExportedResources(ctx, &exportedResources, namespace, errChan, &wg)
+	go in.fetchIstioConfigList(ctx, &istioConfigList, &mtlsDetails, &rbacDetails, namespace, errChan, &wg)
 	go in.fetchServices(ctx, &services, namespace, errChan, &wg)
 	go in.fetchWorkloads(ctx, &workloads, namespace, errChan, &wg)
```
Comment (lucasponce) [inline business/istio_validations.go:160]: I have a similar thought here with the "fetchWorkloads" and "fetchAllWorkloads".
Do we need to keep both?

I know it's technical debt, but I think this work is a good opportunity to improve it.

File: business/istio_validations.go [hunk 3]
```diff
@@ -157,42 +153,40 @@ func (in *IstioValidationsService) GetIstioObjectValidations(ctx context.Context
 	errChan := make(chan error, 1)
 
 	// Get all the Istio objects from a Namespace and all gateways from every namespace
-	wg.Add(9)
+	wg.Add(7)
 	go in.fetchNamespaces(ctx, &namespaces, errChan, &wg)
-	go in.fetchIstioConfigList(ctx, &istioConfigList, namespace, errChan, &wg)
-	go in.fetchExportedResources(ctx, &exportedResources, namespace, errChan, &wg)
+	go in.fetchIstioConfigList(ctx, &istioConfigList, &mtlsDetails, &rbacDetails, namespace, errChan, &wg)
 	go in.fetchServices(ctx, &services, namespace, errChan, &wg)
```
Comment (lucasponce) [inline business/istio_validations.go:159]: This "var services models.ServiceList" is populated but it seems it is not used in the context of the "GetIstioObjectValidations", isn't it?

File: business/istio_validations.go [hunk 4]
```diff
@@ -63,23 +63,20 @@ func (in *IstioValidationsService) GetValidations(ctx context.Context, namespace
...
 	// We fetch without target service as some validations will require full-namespace details
-	go in.fetchIstioConfigList(ctx, &istioConfigList, namespace, errChan, &wg)
-	go in.fetchExportedResources(ctx, &exportedResources, namespace, errChan, &wg)
+	go in.fetchIstioConfigList(ctx, &istioConfigList, &mtlsDetails, &rbacDetails, namespace, errChan, &wg)
 	go in.fetchNamespaces(ctx, &namespaces, errChan, &wg)
 	go in.fetchAllWorkloads(ctx, &workloadsPerNamespace, errChan, &wg)
-	go in.fetchNonLocalmTLSConfigs(ctx, &mtlsDetails, namespace, errChan, &wg)
-	go in.fetchAuthorizationDetails(ctx, &rbacDetails, namespace, errChan, &wg)
+	go in.fetchNonLocalmTLSConfigs(&mtlsDetails, errChan, &wg)
 	go in.fetchServices(ctx, &services, namespace, errChan, &wg)
```
Comment (lucasponce) [inline business/istio_validations.go:80]: I'm reviewing the "GetValidations" method and probably we don't need to keep using the "fetchServices" at this level.
"GetValidations" is used for "IstioConfigList", "NamespaceValidationSummary", "ServiceDetails" and "ServiceUpdate".

Only in these two cases, the "service" parameter is used then no need to use the "fetchServices" unless it's under that condition.

That would potentially improve the other cases that are more common.

File: business/istio_validations.go (hunk 2 above)
Comment (hhovsepy) [reply business/istio_validations.go:160]: Good point, better to merge here.

File: business/istio_validations.go (hunk 1 above)
Comment (hhovsepy) [reply business/istio_validations.go:78]: gree, fetchAllWorkloads could also return namespaces

**Review by lucasponce — DISMISSED** [review]
LGTM
Thanks, @hhovsepy; these changes have improved a couple of nice performance spots.

**Review by lucasponce — APPROVED** [review]

**Comment by prachiyadav** [comment]
E2E and UI test executed successfully.

**Review by skondkar — APPROVED** [review]
LGTM. UI automation and E2E tests are pass. Verified the manual steps:

Screenshot from 2022-02-16 19-24-56

Screenshot from 2022-02-16 19-24-44

Screenshot from 2022-02-16 19-24-32

**Pull request merged**